### PR TITLE
ref(ui): Add automatic render instrumentation

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
Going to use the Sentry application to test profiling automatically creating render spans and sending them into Sentry. This will add a couple kb to the bundle size.

Coming back to this as having React renders show up in performance would help the workflows around frontend performance. This is a simplified version of a previous PR (this one only has the application `Profiler`), but essentially it uses the production profiling to send in render updates above the 60fps threshold to show potentially long renders in React.

#### See more
- https://reactjs.org/docs/profiler.html 
As stated in the gist about the profiler changes: `Performance impact is minor: 4 additional fiber attributes, some additional calls to peformance.now() within profiled subtrees.` Regardless, for now we'll start with just wrapping `App` to automatically catch all renders.

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]
The overhead of this looks [pretty minimal](https://gist.github.com/bvaughn/25e6233aeb1b4f0cdb8d8366e54a3977#gistcomment-3301669). Let's try it

**Review by AbhiPrasad — APPROVED** [review]

**Comment by AbhiPrasad** [comment]
Have a screenshot of what this looks like?

**Comment by k-fish** [comment]
@AbhiPrasad here are some pictures of it picking up render spans on local (I also have `GridEditable` wrapped here since I was looking at why one was 250ms)
Screen Shot 2021-11-30 at 11 20 43 AM

File: static/app/utils/performanceForSentry.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const transaction = getCurrentSentryReactTransaction();
+  if (transaction && actualDuration > MIN_UPDATE_SPAN_TIME) {
+    const now = timestampWithMs();
+    const span = transaction.startChild({
+      description: `<${id}>`,
+      op: `ui.react.${phase}`,
+      startTimestamp: now - actualDuration / 1000,
+      endTimestamp: now,
+    });
+    span.finish();
```
Comment (AbhiPrasad) [inline static/app/utils/performanceForSentry.tsx:24]: You don't have to call `span.finish` if you give an `endTimestamp`.

Comment (k-fish) [reply static/app/utils/performanceForSentry.tsx:24]: Oh really? TIL

**Pull request merged**